lib/utils/compose_ts.ts

```diff
@@ -155,3 +155,70 @@ async function performResolution(
 		}).then(resolve, reject);
 	});
 }
+
+/**
+ * Enforce that, for example, if 'myProject/MyDockerfile.template' is specified
+ * as an alternativate Dockerfile name, then 'myProject/MyDockerfile' must not
+ * exist.
+ * @param projectPath The project source folder (-s command-line option)
+ * @param dockerfilePath The alternative Dockerfile specified by the user
+ */
+export function validateSpecifiedDockerfile(
+	projectPath: string,
+	dockerfilePath: string = '',
+): string {
+	if (!dockerfilePath) {
+		return dockerfilePath;
+	}
+	const { exitWithExpectedError } = require('../utils/patterns');
+	const { isAbsolute, join, normalize, parse, posix } = require('path');
+	const { existsSync } = require('fs');
+	const { stripIndent } = require('common-tags');
+	const { contains, toNativePath, toPosixPath } = MultiBuild.PathUtils;
+
+	// reminder: native windows paths may start with a drive specificaton,
+	// e.g. 'C:\absolute' or 'C:relative'.
```

Comment:
I have tested this PR under Windows and MacOS (balena build / deploy / push), targeting both the cloud builders and a local device (push ipAddress, and build/deploy -h / -p options to use a RPi's docker daemon instead of Docker for Windows).